cfme/services/catalogs/catalog.py

```diff
@@ -1,115 +1,106 @@
 
-item_form = web_ui.Form(
-    fields=[('type_select', "//select[@id='st_prov_type']"),
-            ('name_text', Input("name")),
-            ('description_text', Input("description")),
-            ('display_checkbox', Input("display")),
-            ('add_button', form_buttons.add)])
+    save_button = Button('Save')
+    cancel_button = Button('Cancel')
 
-match_page = partial(match_location, controller='catalog', title='Catalogs')
 
+class CatalogsView(ServicesCatalogView):
+    title = Text("#explorer_title_text")
 
-def _all_catalogs_add_new(_):
-    if BZ(1213863).blocks:
-        sel.pytest.skip("Blocker on the bug 1213863")
-    catalog_tree('All Catalogs')
-    cfg_btn('Add a New Catalog')
+    @property
+    def is_displayed(self):
+        return (
+            self.in_explorer and
+            self.title.text == 'All Catalogs' and
+            self.catalogs.is_opened and
+            self.catalogs.tree.currently_selected == ["All Catalogs"])
+
+
+class AddCatalogView(CatalogForm):
+
+    add_button = Button("Add")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.title.text == "Adding a new Catalog" and
+            self.catalogs.is_opened
+        )
+
+
+class EditCatalogView(CatalogForm):
+
+    save_button = Button('Save')
+    reset_button = Button('Reset')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.title.text == "Editing Catalog{}".format(self.obj.name) and
+            self.catalogs.is_opened
+        )
+
+
+class DetailsCatalogView(ServicesCatalogView):
+
+    @property
+    def is_displayed(self):
+        return (
+            self.title.text == 'Catalog "{}"'.format(self.obj.name) and
+            self.catalogs.is_opened
+        )
 
 
 class Catalog(Updateable, Pretty, Navigatable):
-    """Represents a Catalog"""
-    pretty_attrs = ['name', 'items']
 
     def __init__(self, name=None, description=None, items=None, appliance=None):
+        Navigatable.__init__(self, appliance)
         self.name = name
         self.description = description
         self.items = items
-        Navigatable.__init__(self, appliance=appliance)
-
-    def __str__(self):
-        return self.name
 
     def create(self):
-        navigate_to(self, 'Add')
-        # For element not found exception (To be removed)
-        sel.sleep(5)
-        web_ui.fill(form, {'name_text': self.name,
-                           'description_text': self.description,
-                           'button_multiselect': self.items},
-                    action=form.add_button)
+        view = navigate_to(self, 'Add')
+        view.fill({'name_text': self.name,
+                   'description_text': self.description,
+                   'button_multiselect': self.items})
+        view.add_button.click()
         flash_str = 'Catalog "{}" was saved'
-        flash.assert_success_message(flash_str.format(self.name))
+        view.flash.assert_success_message(flash_str.format(self.name))
```

Comment:
OPTIONAL: I'd prefer to see formatted string and format statement together. readability counts.